File: TabNine/Python/Q11.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        #Given two sorted arrays nums1 and nums2 of size m and n respectively, return the median of the two sorted arrays.
        # The overall run time complexity should be O(log (m+n)).
        #constraints: 1 <= nums1.length <= 1000 and 1 <= nums2.length <= 1000 and -10^6 <= nums1[i], nums2[i] <= 10^6

        if len(nums1) > len(nums2):
            return self.findMedianSortedArrays(nums2, nums1)
        elif len(nums1) < len(nums2):
            return self.findMedianSortedArrays(nums1, nums2)
        else:
            if len(nums1) % 2 == 1:
                return (nums1[len(nums1) // 2] + nums2[len(nums2) // 2]) / 2.0
            else:
                return (nums1[len(nums1) // 2] + nums2[len(nums2) // 2]) / 2.0
            
    

File: Copilot/JavaScript/Q11.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(nums1, nums2) {
    /*Given two sorted arrays nums1 and nums2 of size m and n respectively, return the median of the two sorted arrays.
    The overall run time complexity should be O(log (m+n)).*/

    // Merge the two arrays
    var mergedArray = nums1.concat(nums2);
    mergedArray.sort(function(a, b){return a-b});

    // Find the median
    var median = 0;
    if (mergedArray.length % 2 == 0) {
        median = (mergedArray[mergedArray.length/2] + mergedArray[(mergedArray.length/2)-1]) / 2;
    }
    else {
        median = mergedArray[Math.floor(mergedArray.length/2)];
    }

    return median;
    
};